aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/UploadOptions.java

```diff
@@ -0,0 +1,124 @@
+package com.amazonaws.mobileconnectors.s3.transferutility;
+
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+
+/**
+ * A class that can hold all of the optional parameters (bucket, metadata, cannedAcl, transferListener).
+ */
+public class UploadOptions {
+    private String bucket;
+    private ObjectMetadata metadata;
+    private CannedAccessControlList cannedAcl;
+    private TransferListener listener;
+
+    /**
+     * Constructor.
+     *
+     * @param bucket        The bucket name
+     * @param metadata      The S3 metadata to associate with this object
+     * @param cannedAcl     The canned ACL to associate with this object
+     * @param listener      The listener to attach to transfer observer
+     */
+    public UploadOptions(String bucket, ObjectMetadata metadata, CannedAccessControlList cannedAcl, TransferListener listener) {
+        this.bucket = bucket;
+        this.metadata = metadata;
+        this.cannedAcl = cannedAcl;
+        this.listener = listener;
+    }
+
+    public String getBucket() {
+        return this.bucket;
+    }
+
+    public ObjectMetadata getMetadata() {
+        return this.metadata;
+    }
+
+    public CannedAccessControlList getCannedAcl() {
+        return this.cannedAcl;
+    }
+
+    public TransferListener getTransferListener() {
+        return this.listener;
+    }
+
+    /**
+     * Builder class for UploadOptions
+     */
+    public static class Builder {
+        private String bucket;
+        private ObjectMetadata metadata;
+        private CannedAccessControlList cannedAcl;
+        private TransferListener listener;
+
+        public Builder() {
```

Comment:
I think this can be private, right?  Since customers will use the static `builder()` below?